sotodlib/io/G3tHK.py

```diff
@@ -0,0 +1,163 @@
+    fields = relationship("HKFields", back_populates='hkfeed')
+    filename = db.Column(db.String)
+    global_start_time = db.Column(db.Integer)
+    path = db.Column(db.String)
+    # scanned
+    # provider id
+    # description
+
+
+class HKFields(Base):
+    """This table is named hkfields and serves as a db for all
+    fields inside each HK'ing file in the hkfeeds table.
+
+    Attributes
+    _________
+    feed_id : integer
+        id that points a field back to its corresponding
+        HK file in the hkfeeds table
+    field : string
+        name of HK field in corresponding HK file
+    start : integer
+        start time for each HK field in ctime
+    end : integer
+        end time for each HK field in ctime
+    """
+    __tablename__ = 'hkfields'
+    id = db.Column(db.Integer, primary_key=True)
+    feed_id = db.Column(db.Integer, db.ForeignKey('hkfeeds.id'))
+    hkfeed = relationship("HKFeeds", back_populates='fields')
+    field = db.Column(db.String)
+    start = db.Column(db.Integer)
+    end = db.Column(db.Integer)
+
+
+class G3tHK:
+    def __init__(self, hkarchive_path, db_path=None, echo=False):
+        """
+        Class to manage a housekeeping data archive
+
+        Args
+        ____
+        hkarchive_path : path
+            Path to the data directory
+        db_path : path, optional
+            Path to the sqlite file
+        echo : bool, optional
+            If true, all sql statements will print to stdout
+        """
+        if db_path is None:
+            db_path = os.path.join(hkarchive_path, '_HK.db')
+
+        self.hkarchive_path = hkarchive_path
+        self.db_path = db_path
+        self.engine = db.create_engine(f"sqlite:///{db_path}")
+        Session.configure(bind=self.engine)
+        self.Session = sessionmaker(bind=self.engine)
+        self.session = Session()
+        Base.metadata.create_all(self.engine)
+
+    def load_fields(self, hkarchive_path, hkfile):
+        """
+        Load fields from .g3 file and start and end time for each field.
+
+        Args
+        ____
+
+        hkarchive_path : string
+            Path to data directory
+        hkfile: string
+            Name of .g3 HK file
+
+        returns: list of field names and corresponding start and
+                 end times from HK .g3 files
+        """
+        f = os.path.join(hkarchive_path, hkfile)
+
+        # enact HKArchiveScanner
+        hkas = hk.HKArchiveScanner()
+        hkas.process_file(f)
+
+        arc = hkas.finalize()
+
+        # get fields from .g3 file
+        fields, timelines = arc.get_fields()
+        hkfs = []
+        for key in fields.keys():
+            hkfs.append(key)
+
+        starts = []
+        ends = []
+        for i in range(len(hkfs)):
+            data = arc.simple(hkfs[i])
+            time = data[0]
+            starts.append(time[0])
+            ends.append(time[-1])
+
+        return hkfs, starts, ends
+
+    def add_hkfeeds(self, hkarchive_path, hkfile):
```

Comment:
look at `index_files` and `index_metadata` in `load_smurf` for examples of going through the whole directory. And this is where `min_ctime` and `max_ctime` comes in.